src/info.c

```diff
@@ -0,0 +1,410 @@
+        case 2:
+            ci->protocols |= FIDO_PIN_PROTOCOL_2;
+            break;
+        default:
+            // Just ignore.
+            break;
+    }
+    return FIDO_OK;
+}
+
+
+static int cbor_info_decode_transport(const cb0r_t element, void *arg) {
+    fido_cbor_info_t *ci = (fido_cbor_info_t*)arg;
+
+    if(CBOR_STR_MEMCMP(element, fido_transport_usb)) {
+        ci->transports |= FIDO_TRANSPORT_USB;
+    } else if(CBOR_STR_MEMCMP(element, fido_transport_nfc)) {
+        ci->transports |= FIDO_TRANSPORT_NFC;
+    } else if(CBOR_STR_MEMCMP(element, fido_transport_ble)) {
+        ci->transports |= FIDO_TRANSPORT_BLE;
+    } else if(CBOR_STR_MEMCMP(element, fido_transport_internal)) {
+        ci->transports |= FIDO_TRANSPORT_INTERNAL;
+    }
+    // Platform MUST tolerate unknown values: https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#authenticatorGetInfo
+    return FIDO_OK;
+}
+
+static int cbor_info_decode_algorithm_entry(const cb0r_t key, const cb0r_t value, void *arg) {
+    fido_cbor_info_t *ci = (fido_cbor_info_t*)arg;
+
+    if (!cbor_utf8string_is_definite(key)) {
+        return FIDO_ERR_CBOR_UNEXPECTED_TYPE;
+    }
+
+    if (CBOR_STR_MEMCMP(key, fido_algorithm_key)) {
+        if (value->type != CB0R_INT && value->type != CB0R_NEG) {
+            return FIDO_ERR_CBOR_UNEXPECTED_TYPE;
+        }
+        int32_t alg_identifier = value->type == CB0R_NEG ? ((~value->value) & 0xffffffff) : (value->value & 0xffffffff);
+        switch (alg_identifier) {
+            case COSE_ALGORITHM_EdDSA:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_EdDSA;
+                break;
+            case COSE_ALGORITHM_ES256:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_ES256;
+                break;
+            case COSE_ALGORITHM_ES384:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_ES384;
+                break;
+            case COSE_ALGORITHM_ES512:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_ES512;
+                break;
+            case COSE_ALGORITHM_ES256K:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_ES256K;
+                break;
+            case COSE_ALGORITHM_PS256:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_PS256;
+                break;
+            case COSE_ALGORITHM_RS256:
+                ci->algorithms |= FIDO_CREDENTIAL_GENERATION_RS256;
+                break;
+        }
+    }
+    return FIDO_OK;
+}
+
+static int cbor_info_decode_algorithm(const cb0r_t element, void *arg) {
+    if (element->type != CB0R_MAP) {
+        return FIDO_ERR_CBOR_UNEXPECTED_TYPE;
+    }
+
+    return cbor_iter_map(element, cbor_info_decode_algorithm_entry, arg);
+}
+
+
+static int parse_info_reply_entry(const cb0r_t key, const cb0r_t value, void *arg) {
+    if (key->type != CB0R_INT || key->value > UINT8_MAX) {
+        // Just ignore the entry according to...
```

Comment:
According to?

Reply:
See <https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#message-encoding>

> If map keys are present that an implementation does not understand, they MUST be ignored. Note that this enables additional fields to be used as new features are added without breaking existing implementations.